LDAR_Sim/src/virtual_world/nonfugitive_emissions.py

```diff
@@ -0,0 +1,163 @@
+"""
+------------------------------------------------------------------------------
+Program:     The LDAR Simulator (LDAR-Sim)
+File:        nonfugitive_emissions
+Purpose: The nonfugitive emissions module.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the MIT License as published
+by the Free Software Foundation, version 3.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+You should have received a copy of the MIT License
+along with this program.  If not, see <https://opensource.org/licenses/MIT>.
+
+------------------------------------------------------------------------------
+"""
+
+from datetime import date, timedelta
+from typing import Any
+from math import ceil
+from typing_extensions import override
+from virtual_world.emissions import Emission
+from file_processing.output_processing.output_utils import EmisInfo
+
+
+EXPIRE = "expired"
+
+
+class NonRepairableEmission(Emission):
+    def __init__(
+        self,
+        emission_n: int,
+        rate: float,
+        start_date: date,
+        simulation_sd: date,
+        repairable: bool,
+        tech_spat_cov_probs: dict[str, float],
+        duration: int,
+    ):
+        super().__init__(
+            emission_n,
+            rate,
+            start_date,
+            simulation_sd,
+            repairable,
+            tech_spat_cov_probs,
+        )
+        self._tagged: bool = False
+        self._days_since_tagged: int = 0
+        self._tagged_by_company: str = None
+        self._tagged_by_crew: str = None
+        self._tagging_rep_delay: int = 0
+        self._expiry_date: date = None
+        self._duration: int = duration
+        days_active_b4_sim: int = (simulation_sd - start_date).days
+        self._days_active_b4_sim = days_active_b4_sim if days_active_b4_sim > 0 else 0
+
+    def __reduce__(self):
+        return self._reconstruct_nonfugitive_emission, (self.__dict__,)
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    @classmethod
+    def _reconstruct_nonfugitive_emission(cls, state):
+        instance = cls.__new__(cls)
+        instance.__setstate__(state)
+        return instance
+
+    def get_init_detect_company(self):
+        return self._init_detect_by
+
+    def tagged_today(self) -> bool:
+        return (
+            self._tagged and (self._days_since_tagged == 0)
+        ) and self._tagged_by_company != EXPIRE
+
+    def get_daily_emissions(self) -> float:
+        return self._rate * 86.4
+
+    def expire(self, emis_info: EmisInfo):
```

Comment:
In the spirit of not tying everything to fugitive emissions - should we do the expiry process using something other than tagged?